### DISCUSSION TITLE
MySQL relationship (many)

### DISCUSSION THREAD

**Discussion by hardc0l2e** [Budibase/budibase#2258] [Help]
Can you help me setup the following:

userlist has many contaclist

the configuration in budibase the this dummy table, what should be the column looks like?

If you have an example it'll be very much appreciated.

![20210804_09 17 49 PM-Budibase_-_Google_Chrome](https://user-images.githubusercontent.com/1472421/128187878-f995eed5-aa55-452f-887b-60f4a95b1e93.png)

**Comment by kevmodrome** [comment]
Can you give some more information? What are you trying to do? What are the steps you have taken to get to this point?

**Reply by hardc0l2e** [reply]
One company/supplier with many contact person information.

In table simple form is like:

table supplier (name, address, taxnumber, etc)
table contacts (name, mobile_number, landline_number, email)

so example:

Supplier XYZ, has 5 contact persons

-----------------
By the way, what's the content of "Through" table?

**Comment by hardc0l2e** [comment]
Here's I made some simple diagram

![image](https://user-images.githubusercontent.com/1472421/128198179-509b8ced-9215-4c97-bf32-0fd08884d774.png)

**Comment by shogunpurple** [comment]
Hi @hardc0l2e - through/join tables are only for many to many relationships in SQL.

If you have a direct 1:N mapping with a foreign key like you do (`supplier_id` in the `contacts` table) you don't need a through table. You should: 
- select a "One" relationship on your supplier table
- link to contacts
- Select `supplier_id` as the FK

then that should be it! Let me know if that works for you.

**Reply by hardc0l2e** [reply]
This one works, but I got another problem, I cannot revert the value of supplier_id to NULL? even I unchecked the relationship of that CONTACTS row with SUPPLIER? Is this normal or a bug? The table has no constraints on NOT NULL for that field.

![agZ3qjmGMj](https://user-images.githubusercontent.com/1472421/128272827-0e57f4a9-84d5-4e28-a3b3-a90176e0990d.gif)

**Reply by shogunpurple** [reply]
Hi @hardc0l2e - this looks like it might be an issue with an empty string being set on that field, rather than `NULL`. Looks like a bug - you can track the issue here: https://github.com/Budibase/budibase/issues/2277
